lib/client.js

```diff
@@ -48,6 +48,20 @@ function Client (broker, conn, req) {
 
   this.connDetails = null
 
+  // queue packets received before client fires 'connect' event. Prevents memory leaks on 'connect' event
+  this.parser._queue = []
+
+  this.on('connected', dequeue)
+
+  function dequeue () {
+    var q = this.parser._queue
+    for (var i = 0, len = q.length; i < len; i++) {
+      handle(this, q[i], this._nextBatch)
+    }
+
+    this.parser._queue = null
+  }
```

Comment:
Please make this top-level, outside of this function.

Reply:
You mean the entire function dequeue?